PP-5142 Unify error responses produced by exception mappers

stephencdaly opened the pull request:
- Add ErrorResponse model to unify the contents of error responses. Include in this an ErrorIdentifier, which will always be "GENERIC" for now.
- All error responses created by exception mappers now include a "message" field which is an array of strings.
- Error responses produced by ResponseUtils.java will be modified in a later commit.

## Code review checklist

### Logging

- [ ] only emit log lines at ERROR level which require immediate attention from a support engineer. These will trigger a zendesk alert.

### Documentation

- [ ] Updated README.md for any of the following ?

* Introduced any new environment variables / removed existing environment variable
* Added new API / updated existing API definition

oswaldquek commented on src/main/java/uk/gov/pay/connector/common/model/api/ErrorResponse.java:
```diff
@@ -0,0 +1,43 @@
+    @JsonProperty("message")
+    private final List<String> messages;
+    
+    @JsonProperty("reason")
```
What's the purpose of this `reason` json field? https://payments-platform.atlassian.net/browse/PP-5142 specifies the body of the response to be 
```
{
  "error_identifier": "zero_pence_transaction_not_allowed",
  "message": ["Zero pence transactions not enabled for this gateway account"]
}
``` 
and the RFC at https://docs.google.com/document/d/16PAyL7fqy2CpcTsQ-OB2rsxS99Z3ERbbqSrsvYDXwzI/edit#heading=h.sjuhspjyssr1 doesn't mention it either.

stephencdaly replied:
As we discussed at the story writing, there is an existing "reason" field that is already included for a handful of errors (see ResponseUtils.java `private static Response responseWith(Status status, String message, String reason) {`)

Not planning on using this any further than it is already, but I want all the error responses to use this object so I'm adding the "reason" field for this purpose.

oswaldquek replied:
👍

oswaldquek commented on src/test/java/uk/gov/pay/connector/gatewayaccount/resource/GatewayAccountResourceValidationTest.java:
```diff
@@ -115,7 +115,7 @@ public void shouldReturn400_whenCorporatePrepaidCreditCardSurchargeOperationIsMi
                 .method("PATCH", Entity.json(jsonNode));
         assertThat(response.getStatus(), is(400));
-        String errorMessage = response.readEntity(JsonNode.class).get("errors").get(0).textValue();
+        String errorMessage = response.readEntity(JsonNode.class).get("message").get(0).textValue();
```
I know it's not directly related to the story but could the test assert the "message" field is an array of size 1 too? Just to be more rigorous.

stephencdaly replied:
Yes, might as well - I'll update the tests

stephencdaly commented:
I've updated the GatewayAccountResourceValidationTests to check the size of the "message" array. Also added in assertions on the "error_identifier" while I was at it.
